qiskit_nature/problems/sampling/protein_folding/peptide/chains/main_chain.py

```diff
@@ -0,0 +1,178 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""A class defining the main chain of a peptide."""
+from typing import List, Sequence
+
+from ...exceptions.invalid_side_chain_exception import (
+    InvalidSideChainException,
+)
+from ...exceptions.invalid_size_exception import InvalidSizeException
+from ..beads.main_bead import MainBead
+from .base_chain import BaseChain
+from .side_chain import SideChain
+
+
+class MainChain(BaseChain):
+    """A class defining a main chain of a peptide."""
+
+    def __init__(
+        self,
+        main_chain_len: int,
+        main_chain_residue_sequences: List[str],
```

Comment:
If `main_chain_len == len(main_chain_residue_sequences)` what is the reason for passing these in separately? Or is this not the case?

Reply:
It is indeed a redundancy. The length of the main chain can be inferred from the length of the list. The reason I kept both is that I thought it is easy to make a mistake when providing the list only - skipping one residue in the list is not easy to notice and leads to a main chain shorter than intended. Having `main_chain_len` makes it easy to eyeball it. Let me know what you think...